### ISSUE TITLE
PlayerEnv auto-reseting

### ISSUE THREAD

**Issue by Benjamin-Etheredge** [hsahovic/poke-env#211]
`PlayerEnv.step` auto resets when battles end. It implies that steps can be made after `done` is reached.
I was curious what `gym` says about this and found this warning after stepping past done on a sample enviornment. 

```
UserWarning: WARN: You are calling 'step()' even though this environment has already returned done = True. You should always call 'reset()' once you receive 'done = True' -- any further steps are undefined behavior.
```

Should this behaviour be allowed? It's definitely not a big deal, but it is something to note.

**Comment by hsahovic** [comment]
Thanks for bringing this up. I wasn't aware that this wasn't the expected behavior - I'll get a fix in this week.

**Comment by Benjamin-Etheredge** [comment]
It technically isn't a bug. The behavior is undefined, so you can treat it however you want. 

But I am partial to it throwing an exception in the case of stepping past `done`. I would think doing so is indicative of a bug in the user's algorithm's code.

**Comment by hsahovic** [comment]
Fixed by 0d28f11269e6d703264bde018e653b8dc7023931
